File: src/frontend/services/apiClients/debug.ts
// Copyright (C) 2026 StableLlama
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

/**
 * Defines the debug unit so this responsibility stays isolated, testable, and easy to evolve.
 */

import { DebugLogEntry } from '../apiTypes';
import { fetchJson } from './shared';

export const debugApi = {
  getLogs: async () => {
    return fetchJson<DebugLogEntry[]>(
      '/debug/llm_logs?_t=' + Date.now(),
      undefined,
      'Failed to fetch debug logs'
    );
  },

  clearLogs: async () => {
    return fetchJson<{ status: string }>(
      '/debug/llm_logs?_t=' + Date.now(),
      { method: 'DELETE' },
      'Failed to clear debug logs'
    );
  },
};

File: src/augmentedquill/api/v1/debug.py
# Copyright (C) 2026 StableLlama
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

"""Defines the debug unit so this responsibility stays isolated, testable, and easy to evolve."""

from fastapi import APIRouter
from augmentedquill.services.llm import llm_logging

router = APIRouter(prefix="/debug", tags=["debug"])

router.add_api_route(
    "/llm_logs", endpoint=lambda: llm_logging.llm_logs, methods=["GET"]
)


@router.delete("/llm_logs")
async def clear_llm_logs():
    """Clear the LLM communication logs."""
    llm_logging.llm_logs.clear()
    return {"status": "ok"}
